pandas/sparse/tests/test_sparse.py

```diff
@@ -770,6 +770,32 @@ def test_combine_first(self):
         assert_sp_series_equal(result, result2)
         assert_sp_series_equal(result, expected)
 
+class TestSparseHandlingMultiIndexes(tm.TestCase):
+
+    def setUp(self):
+        miindex = pd.MultiIndex.from_product([["x","y"], ["10","20"]],names=['row-foo', 'row-bar'])
+        micol = pd.MultiIndex.from_product([['a','b','c'], ["1","2"]],names=['col-foo', 'col-bar'])
+        dense_multiindex_frame = pd.DataFrame(index=miindex, columns=micol).sortlevel().sortlevel(axis=1)
+        self.dense_multiindex_frame = dense_multiindex_frame.fillna(value=3.14)
+
+    def test_to_sparse_preserve_multiindex_names_on_columns(self):
+        sparse_multiindex_frame = self.dense_multiindex_frame.to_sparse()
+        self.assertTrue(self.dense_multiindex_frame.columns.equals(sparse_multiindex_frame.columns))
```

Comment:
Okay, I can change that.

 

 

From: Jeff Reback [mailto:notifications@github.com] 
Sent: Saturday, November 14, 2015 1:06 PM
To: pydata/pandas
Cc: Ezekiel Kruglick
Subject: Re: [pandas] BUG GH11600 - MultiIndex column level names lost when to_sparse() called (#11606)

 

In pandas/sparse/tests/test_sparse.py <https://github.com/pydata/pandas/pull/11606#discussion_r44863609> :

> @@ -770,6 +770,32 @@ def test_combine_first(self):
>          assert_sp_series_equal(result, result2)
>          assert_sp_series_equal(result, expected)
>  
> +class TestSparseHandlingMultiIndexes(tm.TestCase):
> +
> +    def setUp(self):
> +        miindex = pd.MultiIndex.from_product([["x","y"], ["10","20"]],names=['row-foo', 'row-bar'])
> +        micol = pd.MultiIndex.from_product([['a','b','c'], ["1","2"]],names=['col-foo', 'col-bar'])
> +        dense_multiindex_frame = pd.DataFrame(index=miindex, columns=micol).sortlevel().sortlevel(axis=1)
> +        self.dense_multiindex_frame = dense_multiindex_frame.fillna(value=3.14)
> +
> +    def test_to_sparse_preserve_multiindex_names_on_columns(self):
> +        sparse_multiindex_frame = self.dense_multiindex_frame.to_sparse()
> +        self.assertTrue(self.dense_multiindex_frame.columns.equals(sparse_multiindex_frame.columns))

this should all be 2 test and just use
assert_sp_frame_equal or dense version to compare versus and expected frame

—
Reply to this email directly or view it on <https://github.com/pydata/pandas/pull/11606/files#r44863609>  GitHub.  <https://github.com/notifications/beacon/ACOkLuICpc4O02DSIfDD_jNsE1yhKZ1lks5pF5m2gaJpZM4Gia9-.gif>